retz-client/src/main/java/io/github/retz/web/feign/Server.java

```diff
@@ -0,0 +1,115 @@
+/**
+ *    Retz
+ *    Copyright (C) 2016 Nautilus Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package io.github.retz.web.feign;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+import feign.Feign;
+import feign.Param;
+import feign.RequestLine;
+import feign.jackson.JacksonDecoder;
+import feign.jackson.JacksonEncoder;
+import io.github.retz.auth.Authenticator;
+import io.github.retz.protocol.LoadAppRequest;
+import io.github.retz.protocol.Response;
+import io.github.retz.protocol.ScheduleRequest;
+import io.github.retz.protocol.data.Application;
+import io.github.retz.protocol.data.Job;
+
+public interface Server {
+
+    @RequestLine("GET /ping")
+    String ping();
+
+    @RequestLine("GET /jobs")
+    Response list();
+
+    default Response schedule(Job job) {
+        return schedule(new ScheduleRequest(job));
+    }
+
+    @RequestLine("POST /job")
+    Response schedule(ScheduleRequest request);
+
+    @RequestLine("GET /job/{id}")
+    Response getJob(@Param("id") int id);
+
+    @RequestLine("GET /job/{id}/file?path={path}&offset={offset}&length={length}")
+    Response getFile(
+            @Param("id") int id,
+            @Param("path") String path,
```

Comment:
I found that there are some cases that the `@JsonProperty` says `required = true` but the `Request` doesn't null-check, e.g. `GetAppRequest` or `LoadAppRequest`, and `UnloadAppRequest`.
Should I null-check or not for the cases?